### ISSUE TITLE
Support user-generated keys

### ISSUE THREAD

**Issue by JoshOrndorff** [shawntabrizi/polkadot-launch#12]
Currently the script allows the relay chain validators to be chosen from `--alice`, `--bob`, etc. A nice feature would be to generate custom keys (or allow the user to supply their own keys). The tricky part comes because you have to restart the node after inserting keys of the grandpa task won't cast votes.

**Comment by shawntabrizi** [comment]
Do i need to restart the node?

I generate the chain spec on the fly with the users we expect:

https://github.com/shawntabrizi/polkadot-launch/blob/master/src/spec.js

So unless there is something else extra about the keys you want to generate, you would just need to add support for a different JSON config across a couple of files.

**Comment by JoshOrndorff** [comment]
Yeah that code you linked for generating the specs is really nice and wouldn't need much modification.

One difference though, is that when you use custom generated keys, you have to insert them to the node via RPC (you can't use `--alice` or something similar). I think that can be handled though.

The part I'm not as sure about is that the grandpa task doesn't start voting until you've inserted the key _and_ restarted the node ([trancking issue](https://github.com/paritytech/substrate/issues/7268)).

To give you some context for why I'm asking about this. I was thinking of using it to spin up a local moonbean testnet quickly. It isn't high on my list, but I hope to do it in the next few months.
